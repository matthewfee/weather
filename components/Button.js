const Button = ({ callback, styles, children, disabled }) => (
  <button
    className={`btn outline-none disabled:text-white/50 focus:outline-1 focus:outline-white  ${styles}`}
    onClick={callback}
    type="button"
    disabled={disabled}
  >
    {children}
  </button>
);

export default Button;
