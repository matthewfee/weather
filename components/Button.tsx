type ButtonProps = {
  callback: () => void;
  styles: string;
  children: React.ReactNode;
  disabled: boolean;
};

const Button = ({ callback, styles, children, disabled }: ButtonProps) => (
  <button
    className={`btn outline-none disabled:text-white/50 focus:outline-1 focus:outline-white  ${styles}`}
    onClick={callback}
    type="button"
    disabled={disabled}
  >
    {children}
  </button>
);

export default Button;
