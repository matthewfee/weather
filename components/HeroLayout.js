import { useState, useEffect } from 'react';
import { nightBackgroundImageString } from '../constants/constants';
import sunny from '../public/backgrounds/sunny.jpg';
import cloudy from '../public/backgrounds/cloudy.jpg';
import rainy from '../public/backgrounds/rainy.jpg';

const HeroLayout = ({ children, isDaytime, weather }) => {
  const [background, setBackground] = useState(`url(${rainy.src})`);

  const getBackground = () => {
    let imageBackground = `url(${sunny.src})`;

    if (weather?.description?.includes('clouds')) {
      console.log('CLOUDY');
      imageBackground = `url${cloudy.src}`;
    }

    if (weather?.description?.includes('rain')) {
      imageBackground = `url${rainy.src}`;
    }

    console.log(imageBackground);

    setBackground(imageBackground);
  };

  useEffect(() => {
    getBackground();
  }, [weather]);

  return (
    <div
      className="hero hero-background min-h-[110vh] w-screen min-w-fit  "
      style={{
        backgroundSize: 'cover',
        backgroundPosition: 'center bottom',
        backgroundImage: background,
      }}
    >
      <div
        className={`hero-overlay ${
          isDaytime ? `opacity-80` : `opacity-80`
        }  transition-all ease-in-out duration-[2000ms] `}
        style={{
          backgroundColor: `${isDaytime ? '#F26B1D' : '#104040'}`,
          backgroundImage: `${nightBackgroundImageString}`,
        }}
      />
      <div
        className={`hero-content text-center text-neutral-content transition-colors ease-in-out duration-[4000ms] h-0 border 

      flex flex-col p-0 w-full max-w-[600px] m-0 rounded-xl shadow-2xl shadow-black/70 ${
        isDaytime ? 'bg-[#09243B]/60' : 'bg-blue-900/60'
      } ${weather ? 'h-[400px] opacity-100 border-black/50' : 'h-0 border-none'} `}
      >
        {children}
      </div>
    </div>
  );
};

// box-shadow: 0px 6px 20px rgba(0, 0, 0, 0.04);
// backdrop-filter: blur(50px); #0788D9

export default HeroLayout;
