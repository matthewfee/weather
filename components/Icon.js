import Image from 'next/dist/client/image';

const Icon = ({ iconID }) => {
  if (!iconID) {
    return null;
  }
  return (
    <Image
      src={`/icons/${iconID}.png`}
      alt="icon"
      width="150px"
      height="150px"
      loading="eager"
      priority
      placeholder=""
      quality={25}
    />
  );
};

export default Icon;
