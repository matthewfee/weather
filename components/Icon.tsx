import Image from 'next/dist/client/image';
import React from 'react';

const Icon = ({ iconID }) => {
  if (!iconID) {
    return null;
  }
  return (
    <Image
      src={`/icons/${iconID}.svg`}
      alt="icon"
      width="60px"
      height="60px"
      loading="eager"
      priority
      quality={25}
    />
  );
};

export default Icon;
