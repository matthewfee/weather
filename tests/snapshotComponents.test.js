import TestRenderer from 'react-test-renderer';
import Button from '../components/Button';
import HeroLayout from '../components/HeroLayout';
import Icon from '../components/Icon';

it('renders correctly', () => {
  const tree = TestRenderer.create(
    <Button
      callback={() => {
        console.log('TEST');
      }}
      styles="btn-primary btn-lg w-28"
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = TestRenderer.create(<HeroLayout isDaytime />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = TestRenderer.create(<Icon iconID="04d" />).toJSON();
  expect(tree).toMatchSnapshot();
});
