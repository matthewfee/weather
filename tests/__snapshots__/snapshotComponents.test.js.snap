// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn btn-primary btn-lg w-28"
  onClick={[Function]}
  type="button"
/>
`;

exports[`renders correctly 2`] = `
<div
  className="hero hero-background h-screen min-h-[110vh] w-screen min-w-fit"
  style={
    Object {
      "backgroundImage": "linear-gradient(to right bottom, rgba(213, 246, 255, 0.5), rgba(3, 76, 140, 1)), url(https://picsum.photos/2000/1400)",
      "backgroundPosition": "center bottom",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="hero-overlay"
  />
  <div
    className="hero-content w-full text-center text-neutral-content max-w-md flex flex-col items-center rounded-md p-0 m-0"
  />
</div>
`;

exports[`renders correctly 3`] = `
<span
  style={
    Object {
      "background": "none",
      "border": 0,
      "boxSizing": "border-box",
      "display": "inline-block",
      "height": "initial",
      "margin": 0,
      "maxWidth": "100%",
      "opacity": 1,
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "width": "initial",
    }
  }
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "padding": 0,
        "width": "initial",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27150%27%20height=%27150%27/%3e"
      style={
        Object {
          "background": "none",
          "border": 0,
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    />
  </span>
  <img
    alt="icon"
    data-nimg="intrinsic"
    decoding="async"
    onError={[Function]}
    onLoad={[Function]}
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="icon"
      data-nimg="intrinsic"
      decoding="async"
      loading="lazy"
      src="/_next/image?url=%2Ficons%2F04d.png&w=384&q=75"
      srcSet="/_next/image?url=%2Ficons%2F04d.png&w=256&q=75 1x, /_next/image?url=%2Ficons%2F04d.png&w=384&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</span>
`;
